### PR TITLE
JS: Add model for shelljs

### PR DESCRIPTION
**Pull request by asger-semmle**
After turning on full type extraction I noticed from false negatives from not modelling `shelljs` methods as TaintedPath sinks.

[Evaluation](https://git.semmle.com/asger/dist-compare-reports/tree/shelljs) on default.slugs is uneventful.

### CODE REVIEW

**Comment by asger-semmle** [comment]
> Consistently using the range pattern is a little verbose

Yeah it might be overkill to use it twice for this model. Should we get rid of `Instance` and just have `Member`?

File: change-notes/1.21/analysis-javascript.md [hunk 1]
```diff
@@ -5,6 +5,7 @@
 * Support for the following frameworks and libraries has been improved:
   - [koa](https://github.com/koajs/koa)
   - [socket.io](http://socket.io)
+  - [shelljs](https://www.npmjs.com/package/shelljs)
```
Comment (esben-semmle) [inline change-notes/1.21/analysis-javascript.md]:
```suggestion
  - [shelljs](https://github.com/shelljs/shelljs)
```

Comment (asger-semmle) [reply change-notes/1.21/analysis-javascript.md]: Why are we linking to github repos instead of the NPM page, again?

Comment (xiemaisi) [reply change-notes/1.21/analysis-javascript.md]: > GitHub repo > npm page ?

I would say `<`

**Comment by xiemaisi** [comment]
> Should we get rid of Instance and just have Member?

If you don't have a concrete need for it at the moment, perhaps we can simplify it for now.

File: change-notes/1.21/analysis-javascript.md (hunk 1 above)
Comment (esben-semmle) [reply change-notes/1.21/analysis-javascript.md]: The github repository is the official definition of the package, in the absence of a proper home page (like https://socket.io has). The npm page is just derived from the repository information, and our models are not tied to npm in any way.

Comment (xiemaisi) [reply change-notes/1.21/analysis-javascript.md]: The npm page looks much more useful to me, and it has a link to the github repo.

Comment (esben-semmle) [reply change-notes/1.21/analysis-javascript.md]: But this is not a matter of usefulness: it is a matter of documenting what we are doing, not how to use a package. I am fine either way, as long as we are consistent.
Should the preference be `homepage > npm > repo`?

Comment (xiemaisi) [reply change-notes/1.21/analysis-javascript.md]: Yes, I would prefer that preference order.

Comment (asger-semmle) [reply change-notes/1.21/analysis-javascript.md]: The NPM page is also the more stable link, as it's not unusual for github repos to be renamed or moved into an org as the project grows.

**Pull request merged**